lib/discourse_translator/guardian_extension.rb

```diff
@@ -22,14 +22,13 @@ def can_detect_language?(post)
   def can_translate?(post)
     return false if !user_group_allow_translate?
 
-    locale = post.detected_locale
-    return false if locale.nil?
-
-    # I18n.locale is a symbol e.g. :en_GB
-    detected_lang = locale.to_s.sub("-", "_")
-    detected_lang != I18n.locale.to_s &&
-      "DiscourseTranslator::#{SiteSetting.translator}".constantize.language_supported?(
-        detected_lang,
-      )
+    if SiteSetting.experimental_topic_translation
+      return false if post.locale_matches?(I18n.locale)
+      return false if post.translation_for(I18n.locale).present?
+      true
+    else
+      return false if post.locale_matches?(I18n.locale)
```

Comment:
Small one but this check is the same regardless of the conditional so we can technically pull this out of the conditional.